examples/cpp/helloworld/README.md

```diff
@@ -256,5 +256,57 @@ main loop in `HandleRpcs` to query the queue.
 For a working example, refer to [greeter_async_server.cc](greeter_async_server.cc).
 
 
+#### Async bi-directional streaming server/client
 
+Bidirectional streaming RPCs allow both client and server to read and write 
+streaming data from/to each other, and this is supported by both synchronous and
+asynchronous API. The asynchronous API allows these actions to take place
+without blocking using the completion queue, just as in the asynchronous code
+examples above.
+One notable difference from the previous section on async RPCs and async
+streaming (bi-directional or otherwise) is the slightly complex nature of
+RPC flows and setup necessary:
 
+* Ensure the server calls `ServerContext::AsyncNotifyWhenDone(&tag)` during the
+  the stream setup so that you listen to notifications when the client is done 
+  with the stream.
+  
+* Ensure that the client calls `WritesDone` to indicate it is done with the
+  stream.
+
+* Ensure that client/server call `Finish` close the stream. This is an 
+  asymmetric API: The server `Finish`es the stream while the client listens
+  to the `Finish` notification generated by the server. This is an important 
```

Comment:
So, this is an idiomatic use but not really as asymmetric as it seems. The client can, for example, directly call Finish without having invoked WritesDone. I think maybe it should be written accordingly. But we can clean that up later.